docs/generate_command_docs.py

```diff
@@ -56,102 +36,59 @@ def get_command_help_output(cmd: str) -> str:
-    index_page = "reference/cli-commands/cli-commands.md"
-    with open(index_page, "r") as file:
-        text = file.read()
-
-    start_index = text.find("```{toctree}")
-    end_index = text.find("```", start_index + 1) + 3
-    cmd_list = "\n".join(f"{cmd[0]} <{cmd[0]}>" for cmd in cmds)
-
-    toc_tree = f"""\
-```{{toctree}}
-:titlesonly:
-:maxdepth: 1
-
-{cmd_list}
-```"""
-
-    text = text[:start_index] + toc_tree + text[end_index:]
-    with open(index_page, "w") as file:
-        file.write(text)
+class Markers:
+    def __init__(self, cmd: str):
+        self.start = f"<!-- START AUTOMATED OUTPUT FOR {cmd} -->"
+        self.end = f"<!-- END AUTOMATED OUTPUT FOR {cmd} -->"
 
 
-def create_file_if_not_exist(filepath: str, cmd: str) -> bool:
-    file_existed = os.path.exists(filepath)
-    if not file_existed:
-        logger.info(
-            "The doc for command %s doesn't exist, creating from the template.", cmd
-        )
-        with open(filepath, "w") as file:
-            file.write(TEMPLATE)
-    return file_existed
-
-
-def generate_help_command_and_output(cmd: str) -> typing.Tuple[str, str]:
+def generate_example(cmd: str, markers: Markers) -> str:
     args = ["help"] if cmd == "help" else [cmd, "--help"]
     help_cmd_str = " ".join(["pebble"] + args)
     go_run_cmd = " ".join(["go", "run", "../cmd/pebble"] + args)
-    help_cmd_output = get_command_help_output(go_run_cmd).strip()
+    help_output = get_command_help_output(go_run_cmd).strip()
 
-    output = f"""\
-<!-- START AUTOMATED OUTPUT -->
+    return f"""\
+{markers.start}
 ```{{terminal}}
 :input: {help_cmd_str}
-{help_cmd_output}
+{help_output}
 ```
-<!-- END AUTOMATED OUTPUT -->"""
-
-    return help_cmd_str, output
+{markers.end}"""
 
 
-def process_command(cmd: str, description: str):
-    logger.info("Processing doc for command %s.", cmd)
+def insert_example(text: str, markers: Markers, example: str) -> str:
+    start_pos = text.find(markers.start)
+    end_pos = text.find(markers.end) + len(markers.end)
+    return text[:start_pos] + example + text[end_pos:]
 
-    file_path = f"reference/cli-commands/{cmd}.md"
-    file_existed = create_file_if_not_exist(file_path, cmd)
 
+def process_commands(cmds: typing.List[str]):
+    file_path = "reference/cli-commands.md"
     with open(file_path, "r") as file:
         text = file.read()
 
-    if AUTOMATED_START_MARKER not in text:
-        logger.info(
-            'The marker for automated doc generation is not found in the "%s" doc, ignore.',
-            cmd,
-        )
-        return
-
-    help_cmd, help_cmd_output = generate_help_command_and_output(cmd)
-    description = f"The `{cmd}` command is used to {description.lower()}."
-
-    if not file_existed:
-        text = text.format(command=cmd, description=description)
+    for cmd in cmds:
+        logger.info("Processing doc for command %s.", cmd)
+        markers = Markers(cmd)
+        if markers.start not in text:
+            logger.info(
```

Comment:
Should we make this an exception now? I'm concerned that if we add a new Pebble command and run this, it'll just log something, whereas that should probably be a hard failure so we don't forget to add it.

Or, better, the script should add it to the right place (though that might be a bit more work).

Reply:
Hmm. Having the script adding a missing command is going to be complex, because it would also need adding to the logical grouping of commands if we wanted to fully automate. I don't think it's worth the effort really - and this is surely going to be a rare situation

I'd rather break the build if there's a command missing from the doc. I'll change the script so that we raise an exception

Reply:
Yep, that sounds fine to me for now -- thanks.